### PR TITLE
[Eligibility Questionnaire] Store the state of the eligibility questionnaire

### PR DESCRIPTION
**Pull request by elenm**
# Pull Request Template

## Issue

Addresses #216 

- [ ] This change addresses the issue in full
- [x] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

The state will be saved as the user navigates throughout the website. The state of the questionnaire is lost when the user refreshes the page. We use [react context](https://www.netlify.com/blog/2020/12/01/using-react-context-for-state-management-in-next.js/) to store the state. I considered storing some of the values in EligibilityRequirements like "bedroomCounts" and "income" as enums, but decided against it for now because the conversion logic was adding a lot of boilerplate to the components. We can either create the appropriate enums in the backend and refactor later or do the necessary conversions on submission of the form.

The change also adds back buttons to the pages in the questionnaire.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Please describe the tests that you ran to verify your changes. Provide instructions so we can review. Please also list any relevant details for your test configuration

- [x] Desktop View
- [x] Mobile View
- [ ] Test A
- [ ] Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes

image

### CODE REVIEW

File: sites/public/pages/eligibility/age.tsx [hunk 1]
```diff
@@ -49,6 +56,12 @@ const EligibilityAge = () => {
         />
       </FormCard>
       <FormCard>
+        <FormBackLink
+          url={eligibilityRoute(CURRENT_PAGE - 1)}
+          onClick={() => {
+            // Not extra actions needed.
+          }}
```
Comment (avaleske) [inline sites/public/pages/eligibility/age.tsx:63]: Do we need the empty `onClick()` for this to work? If not, can we remove it?

File: sites/public/pages/eligibility/bedrooms.tsx [hunk 2]
```diff
@@ -5,34 +5,48 @@ Prompts the user for the number of bedrooms they need.
...
+  const bedroomsOptions = [
     { id: "studio", label: t("eligibility.bedrooms.studio") },
-    { id: "1", label: "1" },
-    { id: "2", label: "2" },
-    { id: "3", label: "3" },
-    { id: "4+", label: "4+" },
+    { id: "oneBdrm", label: "1" },
+    { id: "twoBdrm", label: "2" },
+    { id: "threeBdrm", label: "3" },
+    { id: "fourBdrm", label: "4+" },
```
Comment (avaleske) [inline sites/public/pages/eligibility/bedrooms.tsx:49]: @tomgebauer has asked that the bedrooms filter actually be a "min bedrooms" filter, so it's options will be 0+, 1+, 2+, 3+, etc. If that's different than what was in the spec for the eligibility questionnaire, can you sync with him?

File: sites/public/pages/eligibility/bedrooms.tsx [hunk 3]
```diff
@@ -45,6 +59,12 @@ const EligibilityBedrooms = () => {
         />
       </FormCard>
       <FormCard>
+        <FormBackLink
+          url={eligibilityRoute(CURRENT_PAGE - 1)}
+          onClick={() => {
+            // Not extra actions needed.
+          }}
```
Comment (avaleske) [inline sites/public/pages/eligibility/bedrooms.tsx:66]: ditto here

**Review by avaleske — APPROVED** [review]
I'm excited to see this working! I just had some clarifying questions about the bedrooms filter.

File: sites/public/pages/eligibility/bedrooms.tsx (hunk 2 above)
Comment (elenm) [reply sites/public/pages/eligibility/bedrooms.tsx:49]: Yeah, I can check with him. In the questionnaire, we let them check more than one entry. I also suspect we'll end up removing the bedrooms question from the questionnaire entirely.

File: sites/public/pages/eligibility/age.tsx (hunk 1 above)
Comment (elenm) [reply sites/public/pages/eligibility/age.tsx:63]: It's required in the component unfortunately.

File: sites/public/pages/eligibility/bedrooms.tsx (hunk 3 above)
Comment (elenm) [reply sites/public/pages/eligibility/bedrooms.tsx:66]: Same answer as above.

**Pull request merged**